src/modules/launcher/Plugins/Microsoft.Plugin.Sys/Main.cs

```diff
@@ -0,0 +1,176 @@
+                    IcoPath = $"Images\\logoff.{IconTheme}.png",
+                    Action = c => NativeMethods.ExitWindowsEx(EWX_LOGOFF, 0)
+                },
+                new Result
+                {
+                    Title = Properties.Resources.Microsoft_plugin_sys_lock,
+                    SubTitle = Properties.Resources.Microsoft_plugin_sys_lock_description,
+                    IcoPath = $"Images\\lock.{IconTheme}.png",
+                    Action = c =>
+                    {
+                        NativeMethods.LockWorkStation();
+                        return true;
+                    }
+                },
+                new Result
+                {
+                    Title = Properties.Resources.Microsoft_plugin_sys_sleep,
+                    SubTitle = Properties.Resources.Microsoft_plugin_sys_sleep_description,
+                    IcoPath = $"Images\\sleep.{IconTheme}.png",
+                    Action = c => NativeMethods.SetSuspendState(false, true, true),
+        },
+                new Result
+                {
+                    Title = Properties.Resources.Microsoft_plugin_sys_hibernate,
+                    SubTitle = Properties.Resources.Microsoft_plugin_sys_hibernate_description,
+                    IcoPath = $"Images\\sleep.{IconTheme}.png", // Icon change needed
+                    Action = c => NativeMethods.SetSuspendState(true, true, true),
+                },
+                new Result
+                {
+                    Title = Properties.Resources.Microsoft_plugin_sys_emptyrecyclebin,
+                    SubTitle = Properties.Resources.Microsoft_plugin_sys_emptyrecyclebin_description,
+                    IcoPath = $"Images\\recyclebin.{IconTheme}.png",
+                    Action = c =>
+                    {
+                        // http://www.pinvoke.net/default.aspx/shell32/SHEmptyRecycleBin.html
+                        // FYI, couldn't find documentation for this but if the recycle bin is already empty, it will return -2147418113 (0x8000FFFF (E_UNEXPECTED))
+                        // 0 for nothing
+                        var result = NativeMethods.SHEmptyRecycleBin(new WindowInteropHelper(Application.Current.MainWindow).Handle, 0);
+                        if (result != (uint) NativeMethods.HRESULT.S_OK && result != (uint)0x8000FFFF)
+                        {
+                            var name = "Plugin: " + Properties.Resources.Microsoft_plugin_sys_plugin_name;
+                            var message = $"Error emptying recycle bin, error code: {result}\n" +
+                                            "please refer to https://msdn.microsoft.com/en-us/library/windows/desktop/aa378137";
+                            _context.API.ShowMsg(name, message);
+                        }
+                        return true;
+                    }
+                }
+            });
+            return results;
+        }
+
+		private void UpdateIconTheme(Theme theme)
```

Comment:
outside scope here but this really should be maybe abstracted out as I think most plugins have to do this.  This way we're consistant.

Reply:
May I open another issue/pull request to solve this?